src/03-modality-agnostic-files.md

```diff
@@ -14,6 +14,23 @@ Templates:
 The file `dataset_description.json` is a JSON file describing the dataset.
 Every dataset MUST include this file with the following fields:
 
+{{ MACROS___make_metadata_table(
+   {
+      "Name": "REQUIRED",
+      "BIDSVersion": "REQUIRED",
+      "HEDVersion": "RECOMMENDED",
+      "DatasetType": "RECOMMENDED",
+      "License": "RECOMMENDED",
+      "Authors": "OPTIONAL",
+      "Acknowledgements": "OPTIONAL",
+      "HowToAcknowledge": "OPTIONAL",
+      "Funding": "OPTIONAL",
+      "EthicsApprovals": "OPTIONAL",
+      "ReferencesAndLinks": "OPTIONAL",
+      "DatasetDOI": "OPTIONAL",
+   }
```

Comment:
having such specifications in the document text would be a big step back from toward the target of having a "machine readable" schema.

Why not to add `metadata` key to the corresponding records, e.g. in this case of https://github.com/bids-standard/bids-specification/blob/master/src/schema/top_level_files.yaml#L14 where similar to `entities` of https://github.com/bids-standard/bids-specification/tree/master/src/schema/datatypes to list the 'requirements' for each one of those? and then in the macros just point to what metadata of to render here (e.g. of `top_level_files.dataset_description` here)?  or you can immediately see that this would not work?

Reply:
> having such specifications in the document text would be a big step back from toward the target of having a "machine readable" schema.

I see it as an incremental move forward. It definitely doesn't solve a lot of the problems (e.g., requirement levels, conditional relationships between metadata fields), but to be fair those are the _hard_ problems so I don't feel too bad about it.

> Why not to add metadata key to the corresponding records, e.g. in this case of https://github.com/bids-standard/bids-specification/blob/master/src/schema/top_level_files.yaml#L14 where similar to entities of https://github.com/bids-standard/bids-specification/tree/master/src/schema/datatypes to list the 'requirements' for each one of those? and then in the macros just point to what metadata of to render here (e.g. of top_level_files.dataset_description here)? or you can immediately see that this would not work?

I'll have to try it out, but that sounds like a good plan for the PR _after_ this one. I figure that, since we want to separate the core definitions from the rules of the schema, we can tackle those two pieces in separate stages.

Reply:
I think the best next step for the schema (after this) is to try to identify the types of rules we can expect from the schema, along with any really weird edge cases. That's what I've tried to do in #620, although I worry that my current list isn't exhaustive. Without a list of necessary rules, I'm not sure if we can define them in YAML files in a way that will satisfy everyone.

Reply:
Sure could and should be a multistage process. I am ok with your proposal. I just thought that  it might be nice to just put rules identification for later pr, thus keep description free form, while already allowing for machine readable specification of possible requirement levels and keeping everything under src/schema. But such a move could indeed be just yet another follow up PR to not block this one